### ISSUE TITLE
Should `Style/OneClassPerFile` report offences in `spec/**/*` by default? 

### ISSUE THREAD

**Issue by tagliala** [rubocop/rubocop#15005]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

False positive

### Cop name

Style/OneClassPerFile

### Steps to reproduce the problem

https://github.com/ifad/hawk/pull/87

```
spec/associations_spec.rb:10:1: C: Style/OneClassPerFile: Do not define multiple classes/modules at the top level in a single file.
class Farm < AssociationTestBase
^^^^^^^^^^
spec/associations_spec.rb:19:1: C: Style/OneClassPerFile: Do not define multiple classes/modules at the top level in a single file.
class Animal < AssociationTestBase
^^^^^^^^^^^^
spec/associations_spec.rb:29:1: C: Style/OneClassPerFile: Do not define multiple classes/modules at the top level in a single file.
class Food < AssociationTestBase
^^^^^^^^^^
spec/associations_spec.rb:36:1: C: Style/OneClassPerFile: Do not define multiple classes/modules at the top level in a single file.
class Image < AssociationTestBase
^^^^^^^^^^^
spec/hawk/linker_spec.rb:16:1: C: Style/OneClassPerFile: Do not define multiple classes/modules at the top level in a single file.
class Post
^^^^^^^^^^
```

### Expected behavior

Not to report issue on specs. As a workaround, I've excluded it specifically, but it does not look a good default to me

```
Style/OneClassPerFile:
  Exclude:
    - 'spec/**/*'
```

### Actual behavior

It also reports issues on specs, where custom classes are being added.

What is the best practice in this case?

### RuboCop configuration

```yaml

```

### RuboCop version

```shell
$ rubocop -V
1.85.1 (using Parser 3.3.10.2, rubocop-ast 1.49.0, analyzing as Ruby 3.0, running on ruby 4.0.1) [arm64-darwin25]
  - rubocop-packaging 0.6.0
  - rubocop-performance 1.26.1
  - rubocop-rails 2.34.3
  - rubocop-rake 0.7.1
  - rubocop-rspec 3.9.0
```

**Comment by tagliala** [comment]
Thanks! Of course I also forgot `test/**/*`. Should I open a new issue?

**Comment by koic** [comment]
Minitest assumes a single top-level `class FooTest < ...` definition per file, so it is intentionally not included.

**Comment by tagliala** [comment]
@koic thanks for the clarification.

I'm not using minitest so much, but I still use it on some inherited projects, where I have multiple classes as well, and I was wondering how rails deals with this situation

I've checked some random test files like

https://github.com/rails/rails/blob/main/activesupport/test/tagged_logging_test.rb
https://github.com/rails/rails/blob/main/activesupport/test/test_case_test.rb

and I've noticed they are defining multiple classes as well.

So I've executed this cop on rails main

```sh
$ git rev-parse --short HEAD
8041099b68

$ rubocop --force-default-config --version
1.86.0

$ rubocop --force-default-config --only Style/OneClassPerFile **/*_test.rb
```

Results:

```
1261 files inspected, 956 offenses detected
```

It seems that this is the case in most of the files.

Of course this isn't about whether Rails is 'right' or 'wrong'.

In production, the one-class-per-file rule is a powerful guardrail against monolithic files and spaghetti code. However, in the context of testing, these classes are frequently ephemeral, serving only to isolate a specific edge case or mock a dependency. 

Forcing a structural separation in these cases often introduces unnecessary file-system bloat and obscures the relationship between the test and its supporting helpers.

Given the prevalence of this pattern in such a widely referenced project, Is it possible to reconsidering whether `Style/OneClassPerFile` should remain enabled by default for test directory?

**Comment by koic** [comment]
@tagliala Thanks for the detailed investigation. It seems reasonable to exclude the test directory by default. I opened #15078.
